### PR TITLE
440

### PR DESCRIPTION
**Pull request by ivicac**
- **440 - client - Fix the issue with the broken layout in WorkflowEditor when WorkflowNodesSidebar is open**
- **440 - client - Rename**
- **565 - client - Reorganize**

> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.

## Description

> Please include a summary of the changes and which issue has been fixed. Please also include relevant motivation
> and context. List any dependencies that are required for this change.

Fixes # (issue)

> if no issue exists, please create an issue and ask the maintainers about this first

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- New connector (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+        componentName: workflowConnection.componentName,
+        componentVersion: workflowConnection.componentVersion,
+    });
+
+    return (
+        <div className="flex items-center justify-between">
+            <div className="space-x-1">
+                {componentDefinition && (
+                    <Label>
+                        {`${componentDefinition?.title}`}
```
Comment (kresimir-coko) [inline client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx:74]: dovoljno je samo `{componentDefinition?.title}`

File: client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+                        <TooltipContent>Workflow Connection Key</TooltipContent>
+                    </Tooltip>
+                )}
+            </div>
+
+            <div>
+                <Button className="text-destructive" onClick={onRemoveClick} size="sm" variant="link">
+                    Remove
+                </Button>
+            </div>
```
Comment (kresimir-coko) [inline client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx:95]: `div` viška

File: client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+                                                    name="componentName"
+                                                    onBlur={field.onBlur}
+                                                    onChange={(item) => {
+                                                        const componentDefinition =
+                                                            item?.componentDefinition as ComponentDefinitionBasicModel;
+
+                                                        form.setValue('componentName', componentDefinition.name, {
+                                                            shouldDirty: true,
+                                                        });
+                                                        form.setValue('componentVersion', componentDefinition.version, {
```
Comment (kresimir-coko) [inline client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx:184]: fali prazan red

File: client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+    });
+
+    /* eslint-disable @typescript-eslint/no-non-null-asserted-optional-chain */
+    const {data: connections} = useGetConnectionsQuery(
+        {
+            componentName: componentDefinition?.name!,
+            connectionVersion: componentDefinition?.connection?.version,
+        },
+        !!componentDefinition
+    );
```
Comment (kresimir-coko) [inline client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx:241]: `/* eslint-disable @typescript-eslint/no-non-null-asserted-optional-chain */`

ovaj TS error je u pravu, nemoj ga disableati ovako nego ga rješi

File: client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+            workflowModel: {
+                definition: JSON.stringify(workflowDefinition, null, SPACE),
+                version: workflow.version,
+            },
+        });
+    };
+
+    return (
+        <div className="flex h-full flex-col gap-4 overflow-auto p-4 pt-3.5">
+            <div className="flex  items-center font-semibold">
```
Comment (kresimir-coko) [inline client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx:422]: 2x razmak

File: client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx [hunk 6]
```diff
@@ -0,0 +1,508 @@
...
+                                workflowTestConfigurationConnection={
+                                    workflowTestConfigurationConnections &&
+                                    workflowTestConfigurationConnections.length > 0 &&
+                                    workflowTestConfigurationConnections
+                                        ? workflowTestConfigurationConnections.filter(
+                                              (workflowTestConfigurationConnection) =>
+                                                  workflowTestConfigurationConnection.workflowConnectionKey ===
+                                                  workflowConnection.key
+                                          )[0]
+                                        : undefined
```
Comment (kresimir-coko) [inline client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx:449]: koja kobasica, ako imaš volje izbaci u varijablu

File: client/src/pages/automation/project/components/WorkflowNodesTabs.tsx [hunk 7]
```diff
@@ -47,62 +53,78 @@ const WorkflowNodesTabs = ({
...
-                                />
-                            ))
-                        ) : (
-                            <span className="block px-3 py-2 text-xs text-gray-500">No trigger components found.</span>
-                        )}
-                    </ul>
-                </TabsContent>
-            )}
+            <div>
+                <div className={twMerge('overflow-y-auto absolute', popover && 'h-80', !popover && 'h-full')}>
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/components/WorkflowNodesTabs.tsx:57]: `absolute` nije poželjan ali ću ja pogledat i popravit kad se merga

**Review by kresimir-coko — CHANGES_REQUESTED** [review]

File: client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx (hunk 4 above)
Comment (ivicac) [reply client/src/components/Properties/components/PropertyCodeEditor/PropertyCodeEditorSheetConnectionsSidebar.tsx:241]: Ovo ne mogu trenutno riješiti, ali kod neće puknuti zbog `!!componentDefinition`, koji kaže da li se query mora pozvati ili ne

**Pull request closed without merging**